site/search/searchHooks.ts

```diff
@@ -66,88 +78,22 @@ export function useTagGraphTopics(topicTagGraph: TagGraphRoot | null): {
 }
 
 /**
- * Handles analytics tracking for search state changes
+ * Handles analytics tracking for search state changes.
  */
-export function useSearchStateAnalytics(
+export function useSearchAnalytics(
     state: SearchState,
-    analytics: SiteAnalytics,
-    isInitialUrlStateLoaded: boolean
+    analytics: SiteAnalytics
 ): void {
-    const stateAsUrl = searchStateToUrl(state)
+    // Serialize state to string to use as effect dependency without
+    // triggering on every render
+    const stateKey = useMemo(
+        () => stateToSearchParams(state).toString(),
+        [state]
+    )
```

Comment:
Why do we do this? Because state can be the same object even though some of its inner properties might change, and we want to log analytics in that case? If so, that goes against the React's immutability expectations.

If it's gonna be a different object, then we don't need the `stateKey` to trigger the effect.

Reply:
I see what you mean. You can ignore that intermediary refactor, it's more or less a left-over from the previous state implementation that is addressed in subsequent PRs up the stack (if anything the challenge in this hook is to not trigger too many times).